torch/_dynamo/variables/builder.py

```diff
@@ -1498,6 +1500,135 @@ def wrap_symint(self, value):
+        # ShapeEnv cannot deal with nan
+        if (
+            torch._dynamo.config.specialize_float
+            or is_constant_source(self.get_source())
+            or math.isnan(value)
+        ):
+            self.install_guards(GuardBuilder.CONSTANT_MATCH)
+            return ConstantVariable.create(value=value, source=self.source)
+
+        # NB: We don't do automatic dynamic.  Since we have a guard mechanism,
+        # there isn't really any downside to trying to be dynamic for float
+        # all the time.  Unlike ints, this won't make codegen perf worse.
+        # Modest cost to compile time.
+
+        wrapped_value = torch.tensor(value)
+        # TODO: Switch RandomValueSource over to use this, this is more
+        # accurate
+        assert not isinstance(self.get_source(), RandomValueSource)
+        install_guard(self.get_source().make_guard(GuardBuilder.TYPE_MATCH))
+
+        options = {"source": self.get_source(), "raw_value": value}
+
+        # TODO: Maybe the tensor-ification should be built into the source,
+        # rather than by special pattern match
+        proxy = self.tx.output.root_tracer.create_graph_input(
+            re.sub(r"[^a-zA-Z0-9]+", "_", self.name),
+            type(wrapped_value),
+            source=self.get_source(),
+        )
+
+        unspec_var = wrap_fx_proxy_cls(
+            UnspecializedPythonVariable,
+            tx=self.tx,
+            proxy=proxy,
+            example_value=wrapped_value,
+            **options,
+        )
+        assert isinstance(unspec_var, UnspecializedPythonVariable)
+        self.tx.output.unspec_variable_map[self.name] = unspec_var
+
+        # I guess hypothetically constant source could happen, I kind of want
+        # to handle this totally differently though... no reason to create the
+        # fake tensor in that case
+        assert not is_constant_source(self.get_source())
+
+        if self.tx.export and not isinstance(self.get_source(), LocalSource):
+            raise AssertionError(
+                f"Dynamo attempts to add additional input during export: value={wrapped_value}, source={self.get_source()}"
+            )
+        fake_tensor_value = None
+        example_value = unspec_var.proxy.node.meta["example_value"]
+        assert is_fake(example_value)
+
+        fake_tensor_value = example_value
+        assert fake_tensor_value.fake_mode is self.tx.fake_mode, (
+            f"fake mode ({fake_tensor_value.fake_mode}) from fake tensor metadata doesn't match mode"
+            "({self.tx.fake_mode}) from InstructionTranslator"
+        )
+
+        # There's something a bit incoherent about pass_arg_as_tensor,
+        # specifically regarding sources.
+        #
+        # Specifically, suppose we have "x: float" local argument.  We
+        # eventually end up with an UnspecializedPythonVariable denoting
+        # torch.as_tensor(x)... but it's source is still L['x'] (which if you
```

Comment:
You should create a new source. This is exactly what we do when handling NumPy arrays and we back them by a Tensor.

Reply:
Ah, so concretely, you are suggesting the UnspecializedPythonVariable have something like `torch.as_tensor(L['x'])`, but then the SymNodeVariable (below) has a simplified source just `L['x']`? I suppose this is more accurate. Though, the hope is that you never actually perform guards on the UnspecializedPythonVariable, since it never escapes from this function!

Reply:
Any chance we can put an assert somewhere making sure this is true? I don't know where could it be the best place to put it, or whether there is one...

Reply:
Can we just not put a source on the intermediary Variable?

Reply:
Source is mandatory per asserts

```
  File "/data/users/ezyang/a/pytorch/torch/_dynamo/variables/builder.py", line 303, in __call__
    vt = self._wrap(value)
  File "/data/users/ezyang/a/pytorch/torch/_dynamo/variables/builder.py", line 445, in _wrap
    return type_dispatch(self, value)
  File "/data/users/ezyang/a/pytorch/torch/_dynamo/variables/builder.py", line 1174, in wrap_literal
    return self.wrap_symfloat(value)
  File "/data/users/ezyang/a/pytorch/torch/_dynamo/variables/builder.py", line 1545, in wrap_symfloat
    unspec_var = wrap_fx_proxy_cls(
  File "/data/users/ezyang/a/pytorch/torch/_dynamo/variables/builder.py", line 1817, in wrap_fx_proxy_cls
    assert "source" in options and options["source"] is not None
AssertionError: 
```

Reply:
I add a new source